### PR TITLE
add an admin-only 'internal notes' field on user & org (#2773)

### PR DESCRIPTION
**Pull request by landryb**
took me less than 2h, works fine here in very basic testing, ie:
* i can set an org note directly in ldap
* its shown in the console UI
* if i modify it an event is logged,
* and the new value is saved into ldap

To modify an existing `cn=config` georchestra schema i used this ldif:
```
$cat ~/add_knownedgeInfo.ldif 
dn: cn={6}georchestra,cn=schema,cn=config
changetype: modify
delete: olcObjectClasses
-
add: olcObjectClasses
olcObjectClasses: ( 1.3.6.1.4.1.53611.1.1.1 NAME 'georchestraUser' DESC 'georchestra user' SUP top AUXILIARY MAY privacyPolicyAgreementDate )
olcObjectClasses: ( 1.3.6.1.4.1.53611.1.1.2 NAME 'georchestraOrg' DESC 'georchestra org' SUP top AUXILIARY MAY (jpegphoto $ labeledURI $ knowledgeInformation ))

$sudo ldapmodify -Y EXTERNAL -H ldapi:/// -f ~/add_knownedgeInfo.ldif
```

i chose to add the attr to orgExt/georchestraOrg deliberately, and reused the existing/deprecated attribute `knowledgeInformation` (cf https://docs.ldap.com/specs/rfc2256.txt) as i think it properly conveys the meaning of 'knownledge about this org', and from my understanding its attr type allows 32k chars storage.

Hoping this can of course make 19.12..

after all, this is all about scratching your own itch, and do it yourself ;) cc @pierrejego

### CODE REVIEW

**Comment by landryb** [comment]
And this would of course fix #2773 ;)

**Comment by landryb** [comment]
travis of course might choke on new attributes but i dunno what needs to be done to make it use the newer schema.

**Comment by pierrejego** [comment]
>   this is all about scratching your own itch, and do it yourself ;) cc @pierrejego

You know that if travis CI failed it can not be merged, this is part of the PR :stuck_out_tongue:

I will have a look tomorrow to test it. I would like to see how you manage to hide user note to user.

**Comment by landryb** [comment]
> I would like to see how you manage to hide user note to user.

it's not 'hidden', just not displayed ? the account/details form that regular users have access to doesnt display a textarea for it, neither do the account/org creation form, and afaict you need ROLE_SUPERUSER to issue a GET on `/console/private/orgs/<orgid>` which returns the notes attribute.

**Comment by landryb** [comment]
Oh and `console/account/orgdetails` (which a user has access to modify his org details) doesnt display the field either, so to me it's "hidden" to a regular user.

**Comment by landryb** [comment]
@fvanderbiest is the use of `knowledgeInformation` fine ? May i add the same field to `georchestraUser` / the account object ?

File: console/src/main/webapp/manager/app/assets/lang/es.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
   "org.description"       : "Descripción",
   "org.url"               : "Website",
   "org.logo"              : "Logo",
+  "org.note"              : "Notes",
```
Comment (fvanderbiest) [inline console/src/main/webapp/manager/app/assets/lang/es.json]: Notas ?

File: console/src/main/webapp/manager/app/assets/lang/de.json [hunk 2]
```diff
@@ -109,6 +109,7 @@
   "org.description"       : "Beschreibung",
   "org.url"               : "Webseite",
   "org.logo"              : "Logo",
+  "org.note"              : "Notes",
```
Comment (fvanderbiest) [inline console/src/main/webapp/manager/app/assets/lang/de.json]: Aufzeichnungen ?

File: console/src/main/webapp/manager/app/assets/lang/fr.json [hunk 3]
```diff
@@ -109,6 +109,7 @@
   "org.description"       : "Description",
   "org.url"               : "Site web",
   "org.logo"              : "Logo",
+  "org.note"              : "Notes",
```
Comment (fvanderbiest) [inline console/src/main/webapp/manager/app/assets/lang/fr.json]: Missing NL translation

**Comment by fvanderbiest** [comment]
> is the use of `knowledgeInformation` fine ? May i add the same field to `georchestraUser` / the account object ?

Sounds like a neat idea ! Thanks for the contrib

File: console/src/main/webapp/manager/app/assets/lang/fr.json (hunk 3 above)
Comment (landryb) [reply console/src/main/webapp/manager/app/assets/lang/fr.json]: on a sidenote nl.json lacks at least org.description, org.url, org.logo, org.cities, org.orgType & imageinput.label...

Comment (fvanderbiest) [reply console/src/main/webapp/manager/app/assets/lang/fr.json]: Indeed.
Do you want me to reformat it ?

Comment (landryb) [reply console/src/main/webapp/manager/app/assets/lang/fr.json]: i'll merge/add the missing keys when adding the new ones..

Comment (fvanderbiest) [reply console/src/main/webapp/manager/app/assets/lang/fr.json]: Done with 0d0405374a99167d5aafcbcddc91758158858138

Comment (fvanderbiest) [reply console/src/main/webapp/manager/app/assets/lang/fr.json]: ping @bendeg

Comment (landryb) [reply console/src/main/webapp/manager/app/assets/lang/fr.json]: c7ab0bb should take care of most of the missing ones, of course to be translated from en to nl.

**Comment by landryb** [comment]
travis will probably still choke on the tests because of the new ldap attrs in the ldif that arent understood by the schema shipped by the latest docker ldap image (i suppose so) but i've tested it locally and it works, editing an existing knowledgeInformation attribute on a `georchestraUser` object, setting an empty value removed the attr, and adding a value created the attr.

To be discussed: where should the new field go in the user/org edit form, and i'm not very happy with 43a4c1c but the way UserSchema works, it seems its the simplest way to do, or i can rename all `user.note` mentions/translation keys to `user.knowledgeInformation` but that's a bit wide when editing the json translations.

Other than that, please review the code ... and test ;)

**Comment by landryb** [comment]
oh and maybe the label could be 'Notes internes' / 'Internal notes' to clearly convey the internal status of the value :)

**Comment by fvanderbiest** [comment]
> To modify an existing `cn=config` georchestra schema i used this ldif

OK, the PR looks good to me, but it should also update the release notes + upgrading guide.
Thanks !

**Comment by landryb** [comment]
> > To modify an existing `cn=config` georchestra schema i used this ldif
> 
> OK, the PR looks good to me, but it should also update the release notes + upgrading guide.
> Thanks !

i dont think the upgrading guide/doc needs an update, since in the same release we're adding the `georchestraUser` & `georchestraOrg` objectClasses that werent present in previous releases, the existing procedure you documented in https://github.com/georchestra/georchestra/tree/master/migrations/19.12 will take care of it - but i just noticed right now that maybe you're referring to https://github.com/georchestra/georchestra/blob/master/migrations/19.12/upgrade_ldap_from_19.04_to_19.12.ldif which is more or less a copypaste of https://github.com/georchestra/georchestra/blob/master/ldap/docker-root/georchestraSchema.ldif ? should this ldif be updated ?

**Comment by fvanderbiest** [comment]
> should this ldif be updated ?

:+1:

**Comment by fvanderbiest** [comment]
Issue with translation ...
Capture du 2020-02-03 10-53-44

Also, "Notes" in the forms should be renamed into "Internal notes", to make it clearer.
Hoping admins will not fall into [Code pénal - Article 226-19](https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000026268247&cidTexte=LEGITEXT000006070719)

**Comment by landryb** [comment]
> > should this ldif be updated ?
> 
> 👍

cf ffc391a

**Comment by landryb** [comment]
> Issue with translation ...
> Capture du 2020-02-03 10-53-44

fixed by bd9e6aa77
> 
> Also, "Notes" in the forms should be renamed into "Internal notes", to make it clearer.

Fixed by d8275aa 


> Hoping admins will not fall into [Code pénal - Article 226-19](https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000026268247&cidTexte=LEGITEXT000006070719)

we provide tools, we're not responsible for users shooting themselves in the foot :)

**Comment by fvanderbiest** [comment]
> Fixed by [d8275aa](https://github.com/georchestra/georchestra/commit/d8275aa9f215379dd4f8dc27fb29661b4f24fc19)

Thanks.
I'm not a big fan of "Internal Notes" with a capital N for the second word (in every lang, except maybe for German (which you did not update ?).

**Comment by landryb** [comment]
i'll leave that to someone with german knowledge :)

**Comment by landryb** [comment]
8676de3ed lowercased the second word if you want to cherrypick it.. thanks for merging anyway :)

**Comment by fvanderbiest** [comment]
related: @cmangeat fixed b597b0da4a0d39c5f6c691beb2755b3f2e820db6 on `20.0.x` and I ported it to `master`

**Pull request merged**